apps/web/src/lib/functions/file-loaders/epub/generate-epub-html.ts

```diff
@@ -139,32 +139,41 @@ export default function generateEpubHtml(
       htmlHref = itemIdToHtmlRef[itemIdRef];
     }
 
-    const regexResult = /.*<body(?:[^>]*id="(?<id>.+?)")*[^>]*>(?<body>(.|\s)+)<\/body>.*/.exec(
-      data[htmlHref] as string
-    )!;
+    const contentParser = new DOMParser();
+    const parsedContent = contentParser.parseFromString(data[htmlHref] as string, 'text/html');
 
-    const bodyId = regexResult?.groups?.id || '';
-    let innerHtml = regexResult?.groups?.body || '';
+    const htmlClass = parsedContent.querySelector('html')?.className || '';
+
+    const body = parsedContent.querySelector('body');
```

Comment:
body is a propery on htmldocument and can be destructured from it - no need to query it